design-documents/static-checks-for-templates.md

```diff
@@ -0,0 +1,27 @@
+### Overview
+
+At the time of this writing only 85% of the `*.phtml` files in Magento core are being statically 
+tested due to `@codingStandardsIgnore` present in over 700 templates. There is a current security initiative 
+that requires a large number of templates to be modified and they need to be covered by static checks moving 
+forward as part of the Acceptance Criteria of the initiate. 
+
+However, due to the lack of current static template checks the codebase has thousands of static template errors and will require many 
+hours to address every issue in every file. Additionally, many of the tests are not reasonable, checking incorrectly, or simply gratuitous for `*.phtml` templates. 
+
+
+### Design
+
+To address this issue for the immediate efforts as well as for all developers in the future I am proposing to remove the following sniffs
+for `*.phtml` files.
+
+|Sniff|Rationale|
+|-----|---------|
+| `Magento2.Files.LineLength` |HTML is frequently much cleaner when kept on single lines even if they are longer. This sniff also incorrectly counts lines containing PHP.|
```

Comment:
It may mean that there are too much nesting levels. Such violations may be fixed with automated tooling.

Needs to be provided some insight on current situation to decide.

Reply:
We have no standards for HTML nesting level requirements that we in any way claim to follow. Therefore the comment about "too much nesting levels" seems like a personal opinion rather than a requirement. Please share with me any requirement if there is one documented. All of the automated tooling works for PHP or HTML but I've never seen anything that works for both. Additionally, our existing tooling only works for PHP.

Reply:
That is not what I was saying :) I didn't say a word regarding nesting level enforcement.

1) Line length is a good limitation to start from as high nesting level by implication leads to longer lines
2) I did some experiments before with automated code formatting by setting absolute line limit to 80 or 120 characters and it worked smoothly

Reply:
Here's a very common example of cases where line length is arbitrary and would be messy to try and break it apart:

```php
                                <label class="admin__field-label"
                                       for="sidebar-<?= $block->escapeHtmlAttr($block->getSidebarStorageAction()) ?>-<?= $block->escapeHtmlAttr($block->getIdentifierId($_item)) ?>">
                                </label>
                        </td>
                    </tr>
                </tbody>
            </table>
        </div>
    </div>
</div>
```

I do share the sentiment that every error could be addressed in some way however it doesn't make too much sense to do it. High levels of nesting in PHP are a code smell and a sign that it can be refactored to have less code paths and run more efficiently. Similarly, long lines have a similar idea. However, with HTML nesting is a natural part of the language and there at most there can be a mild correlation between nesting level and good design. So, forcing arbitrary constraints on PHP because of HTML structure is where I disagree with this approach. 

Also, to reiterate what I've said elsewhere, the purpose of this PR is to make templates easier to write and refactor without risking too much change. I understand that in some cases this may come at cost of slightly longer lines but engineering is all about finding reasonable compromises and solutions.